process/smartContract/builtInFunctions/saveUserName.go

```diff
@@ -85,11 +85,12 @@ func (s *saveUserName) ProcessBuiltinFunction(
 		vmOutput.OutputAccounts = make(map[string]*vmcommon.OutputAccount)
 		setUserNameTxData := core.BuiltInFunctionSetUserName + "@" + hex.EncodeToString(vmInput.Arguments[0])
 		outTransfer := vmcommon.OutputTransfer{
-			Value:     big.NewInt(0),
-			GasLimit:  vmInput.GasProvided,
-			GasLocked: vmInput.GasLocked,
-			Data:      []byte(setUserNameTxData),
-			CallType:  vmcommon.AsynchronousCall,
+			Value:         big.NewInt(0),
+			GasLimit:      vmInput.GasProvided,
+			GasLocked:     vmInput.GasLocked,
+			Data:          []byte(setUserNameTxData),
+			CallType:      vmcommon.AsynchronousCall,
+			SenderAddress: vmInput.CallerAddr,
```

Comment:
no need for backwardsCompatibility flag here?

Reply:
it is already put inside smart contract processor.